matter_persistence/redis/utils.py

```diff
@@ -0,0 +1,15 @@
+import gzip
+import pickle
+from typing import Any
+
+
+def compress_pickle_data(data: Any) -> bytes:
+    pickled_data = pickle.dumps(data)  # Serialize the data into bytes using pickle
+    compressed_data = gzip.compress(pickled_data)  # Compress the pickled data using gzip
+    return compressed_data
+
+
+def decompress_pickle_data(compressed_data: bytes) -> Any:
+    decompressed_data = gzip.decompress(compressed_data)  # Decompress the compressed data using gzip
+    data = pickle.loads(decompressed_data)  # Deserialize the data back into Python objects using pickle
+    return data
```

Comment:
Can we please get rid of pickling? That's considered unsafe. I'd much rather store the info on the class that's supposed to be serialised / deserialised & limit caching to only `pydantic` models.

Reply:
I have nothing against restricting caching to pydantic objects.

Reply:
After a discussion with Tomer, pickling is re-introduced.